### DISCUSSION TITLE
Is there a way to use type inference with useRouteData?

### DISCUSSION THREAD

**Discussion by cill-i-am** [sergiodxa/remix-utils#96] [Q&A] [answered]
I'm trying to rely on inference over passing interfaces all over the place. And i'm wondering if theres a way to make this work: `useRouteData<typeof getAvailableHealthPlansLoader>`

It's returning a Promise<TypedResponse>, so i'm half way there. But awaiting here doesn't work

**Comment by sergiodxa** [comment, marked as answer]
You need to use the Remix’s SerializeFrom type to infer the result of the loader

**Reply by lili21** [reply]
Could you provide an example? how to use that?

**Reply by sergiodxa** [reply]
```ts
let routeData = useRouteData<SerializeFrom<typeof loader>>("route/id")
```
